### PR TITLE
String array lifetime

### PR DESCRIPTION
**Pull request by jl-wynen**
Same idea as #1824 but applied to nested scalars.

### CODE REVIEW

File: python/bind_data_access.h [hunk 1]
```diff
@@ -192,22 +192,31 @@ template <class... Ts> class as_ElementArrayViewImpl {
...
-        [&view, &obj](const auto &data) {
+        [&view](const auto &data) {
+          const auto get_parent = [](auto &&view_) {
+            if constexpr (std::is_same_v<std::decay_t<decltype(view)>,
+                                         scipp::Variable>) {
+              return py::cast(view_.data_handle());
+            } else {
+              return py::cast(view_.data().data_handle());
+            }
+          };
```
Comment (SimonHeybrock) [inline python/bind_data_access.h:203]: Can we reuse the helper we added in the other PR?

File: python/bind_data_access.h [hunk 2]
```diff
@@ -261,11 +271,12 @@ template <class... Ts> class as_ElementArrayViewImpl {
...
     } else {
       // Returning reference to element in variable. Return-policy
-      // reference_internal keeps alive `obj`. Note that an attempt to
+      // reference_internal keeps alive `parent`. Note that an attempt to
       // pass `keep_alive` as a call policy to `def_property` failed,
       // resulting in exception from pybind11, so we have handle it by
       // hand here.
-      return py::cast(scalar, py::return_value_policy::reference_internal, obj);
+      return py::cast(scalar, py::return_value_policy::reference_internal,
+                      std::move(parent));
```
Comment (SimonHeybrock) [inline python/bind_data_access.h:279]: Does this affect only strings or does it fix the same issue for nested POD types?

Comment (jl-wynen) [reply python/bind_data_access.h:279]: Do we have any POD types that can be stored in numpy arrays?
I have not tested with types that require ElementArrayView.

File: python/bind_data_access.h (hunk 1 above)
Comment (jl-wynen) [reply python/bind_data_access.h:203]: Sure

File: python/bind_data_access.h (hunk 2 above)
Comment (SimonHeybrock) [reply python/bind_data_access.h:279]: I mean `double` etc?

Comment (jl-wynen) [reply python/bind_data_access.h:279]: This was covered by #1824
But I will add an analogous test to the new one to make sure.

Comment (jl-wynen) [reply python/bind_data_access.h:279]: Added a test with int64 and it passes.

File: python/bind_data_access.h (hunk 1 above)
Comment (jl-wynen) [reply python/bind_data_access.h]: Like this?

**Review by SimonHeybrock — APPROVED** [review]

**Pull request merged**